string name
float32 value